x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -157,6 +157,27 @@ public Object visitNumericArrayLiteral(EsqlBaseParser.NumericArrayLiteralContext
         return new Literal(source, mapNumbers(numbers, (no, dt) -> no.intValue()), DataTypes.INTEGER);
     }
 
+    @Override
+    public Literal visitNumericValue(EsqlBaseParser.NumericValueContext ctx) {
+        Literal result = (Literal) super.visitNumericValue(ctx);
+        if (ctx.MINUS() == null) {
+            return result;
+        }
+        if (result.dataType() == DataTypes.INTEGER) {
+            return new Literal(result.source(), -(int) result.value(), result.dataType());
+        }
+        if (result.dataType() == DataTypes.LONG) {
+            return new Literal(result.source(), -(long) result.value(), result.dataType());
```

Comment:
Good point Bogdan, and the answer is no for both.
Unfortunately the problem is more general, it happens also with EVAL and expressions in general, and I think it's due to the fact that `integer/decimalValue` are parsed without the sign. Checking it now, I think I'll have to go with what @alex-spies suggested

Reply:
Unfortunately they are not, I was misled by `Math.exact()` as well, that's why I went with that solution.
The problem happens at a lower level, when integerValue gets converted from a string to a number it doesn't have a sign (yet), so in case of overflow it's automatically widened to a long or double.
The fix seems to be easier than expected, I'm running the tests now and then I'll push it

Reply:
Fix pushed.
I reverted all the code changes in ExpressionBuilder, since everything was already well managed; only the parser needed to be aware of the sign